pkg/controller/deployment/deployment_controller.go

```diff
@@ -116,7 +116,15 @@ func (r *ReconcileDeployment) Reconcile(request reconcile.Request) (reconcile.Re
 	}
 
 	if !inject.Desired(dep, ns) {
```

Comment:
I would add this cleaning logic on the else statement of the `if inject.Needed(dep, ns) {`

Here: https://github.com/jaegertracing/jaeger-operator/pull/1508/files#diff-c553184997fa4c9f63e697af2f39eb2e6fb43eeff64a0ef66c9e955f61347b02R172